src/main/java/com/csharks/reportingservice/service/ReportingServiceClass.java

```diff
@@ -43,142 +44,167 @@ public ReportingServiceClass(LeadServiceProxy leadServiceProxy, AccountServicePr
             report = new ReportDTO("Minimum Employee Count", findMinEmployeeCount());
-        } else if (reportType.equals("MEAN")) {
+        } else if (reportType.equalsIgnoreCase("MEAN")) {
             report = new ReportDTO("Mean Employee Count", findMeanEmployeeCount());
         }
         return report;
     }
 
     public ReportDTO reportByProductQuantity(String reportType) {
         ReportDTO report = null;
-        if (reportType.equals("MEDIAN")) {
+        if (reportType.equalsIgnoreCase("MEDIAN")) {
             report = new ReportDTO("Median Product Count", findMedianProductQuantity());
-        } else if (reportType.equals("MAX")) {
+        } else if (reportType.equalsIgnoreCase("MAX")) {
             report = new ReportDTO("Maximum Product Count", findMaxProductQuantity());
-        } else if (reportType.equals("MIN")) {
+        } else if (reportType.equalsIgnoreCase("MIN")) {
             report = new ReportDTO("Minimum Product Count", findMinProductQuantity());
-        } else if (reportType.equals("MEAN")) {
+        } else if (reportType.equalsIgnoreCase("MEAN")) {
             report = new ReportDTO("Mean Product Count", findMeanProductQuantity());
         }
         return report;
     }
 
+    // TODO - Fix this method. It is returning the ** product quantity by account instead of ** opportunities.
     public List<ReportDTO> reportOppsNumbersByAccount(String dataType) {
-        List<Account> accounts = getAllAccounts();
+        List<AccountDTO> accounts = getAllAccounts();
         List<ReportDTO> report = new ArrayList<>();
-        if (dataType.equals("MEDIAN")) {
-            for (Account account : accounts) {
+        if (dataType.equalsIgnoreCase("MEDIAN")) {
+            for (AccountDTO account : accounts) {
                 report.add(new ReportDTO("Median Opportunities By Account", findMedianOppsByAccount(account.getId())));
             }
-        } else if (dataType.equals("MAX")) {
-            for (Account account : accounts) {
+        } else if (dataType.equalsIgnoreCase("MAX")) {
+            for (AccountDTO account : accounts) {
                 report.add(new ReportDTO("Maximum Opportunities By Account", findMaxOppsByAccount(account.getId())));
             }
-        } else if (dataType.equals("MIN")) {
-            for (Account account : accounts) {
+        } else if (dataType.equalsIgnoreCase("MIN")) {
+            for (AccountDTO account : accounts) {
                 report.add(new ReportDTO("Minimum Opportunities By Account", findMinOppsByAccount(account.getId())));
             }
-        } else if (dataType.equals("MEAN")) {
-            for (Account account : accounts) {
+        } else if (dataType.equalsIgnoreCase("MEAN")) {
+            for (AccountDTO account : accounts) {
                 report.add(new ReportDTO("Mean Opportunities By Account", findMeanOppsByAccount(account.getId())));
             }
         }
         return report;
     }
```

Comment:
I think this method is not working properly, and we can fix it by changing a query in opportunity.